File: SkypeMessagesExporter/SkypeMessagesExporter/SkypeDatabase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using System.Data.SQLite;
using System.Configuration;

namespace SkypeMessagesExporter
{
    // ta klasa do totalnej refaktoryzacji -> SkypeDatabaseConnector
    // wydzielic pobieranie danych do -> SkypeDatabaseRepository(SkypeDatabaseConnector)
    public class SkypeDatabase
    {
        string skypeDatabaseFile = ConfigurationManager.AppSettings["SkypeDatabaseFile"];
        string skypeDatabaseConnection = ConfigurationManager.AppSettings["SkypeDatabaseConnection"];
        string tempDatabaseFile = ConfigurationManager.AppSettings["TempDatabaseFile"];
        
        public SkypeDatabase()
        {
            // OpenConnection();
        }

        public void ActualiseTemp()
        {
            if (File.Exists(skypeDatabaseFile))
            {
                File.Copy(skypeDatabaseFile, tempDatabaseFile, true);
            }
            else
            {
                throw new Exception("[SkypeDatabase] Database with messages from Skype does not exists");
            }

            /// Disconnect()
        }

        // poprawic tak aby za kazdym razem nie bylo tworzone polaczenie do bazy danych
        public SQLiteConnection GetConnectionToDatabase()
        {
            /*if (dbConnection)
            {
                return dbConnection;
            }
            else
            {
                // OpenConnection();
            }*/

            SQLiteConnection dbConnection = new SQLiteConnection(skypeDatabaseConnection);
            dbConnection.Open();
            return dbConnection;
        }

        private string PrepareQueryForGettingNewMessages(ChannelList channelList)
        {
            string channelsConditions = "";

            channelList.GetAll().ForEach(delegate(Channel channel)
            {
                channelsConditions += " OR (m.convo_id = " + channel.Id + " AND m.id > " + channel.LastMessageId + ")";
            });

            // limit wrzucic do konfiga
           return "SELECT m.id, m.convo_id, c.displayname as convo_name, m.author, m.timestamp, m.body_xml "
                + "FROM Messages m JOIN Conversations c ON (c.id = m.convo_id) WHERE 1 != 1"
                + channelsConditions + " ORDER BY m.id ASC LIMIT 30";
        }

        // przeniesc do SkypeDatabaseRepository -> GetNewMessagesForChannels(ChannelList)
        public List<Message> GetNewMessages(ChannelList channelList)
        {
            List<Message> messages = new List<Message>();
            // polaczenie powinno byc nawiazane wczesniej,
            // teraz uzywac GetConnection() -> if (!connection) Connect() else dbConnection
            SQLiteCommand command = new SQLiteCommand(PrepareQueryForGettingNewMessages(channelList), GetConnectionToDatabase());
            SQLiteDataReader result = command.ExecuteReader();

            while (result.Read())
            {
                Console.WriteLine("- #" + Convert.ToInt32(result["id"].ToString()));

                Message message = new Message(
                    Convert.ToInt32(result["id"].ToString()),
                    Convert.ToInt32(result["convo_id"].ToString()),
                    Convert.ToString(result["convo_name"].ToString()),
                    Convert.ToString(result["author"].ToString()),
                    Convert.ToString(result["timestamp"].ToString()),
                    Convert.ToString(result["body_xml"].ToString())
                );

                messages.Add(message);
            }

            return messages;
        }

        // SkypeDatabaseRepository -> GetChannelList
        public List<Channel> GetChannels()
        {
            string queryToGetChannels = "SELECT id, displayname FROM Conversations ORDER BY displayname";
            List<Channel> channels = new List<Channel>();
            SQLiteCommand command = new SQLiteCommand(queryToGetChannels, GetConnectionToDatabase());
            SQLiteDataReader result = command.ExecuteReader();

            while (result.Read())
            {
                Channel channel = new Channel(
                    Convert.ToInt32(result["id"].ToString()),
                    Convert.ToString(result["displayname"].ToString())
                );

                channels.Add(channel);
            }

            return channels;
        }

    }
}

File: se/skype_database.py
import shutil
import sqlite3

class Skype_Database:
    def __init__(self, original, temp):
        self.skype_database_original = original
        self.skype_database_temp = temp
        
    def refresh(self):
        try:
            shutil.copyfile(self.skype_database_original, self.skype_database_temp)
            return True
        except:
            return False

    def get_new_messages(self, channels, messages_limit):
        messages = []

        channels_query = ""
        channels_id = list(channels.keys())
        for channel_id in channels_id:
            channels_query += " OR (m.convo_id = " + str(channel_id) +  " AND m.id > " + str(channels[channel_id]) + ")";
        
        connection = sqlite3.connect(self.skype_database_temp)
        result = connection.cursor()
        result.execute(
            "SELECT m.id, m.author, m.timestamp, m.body_xml, m.convo_id, c.displayname as convo_name "
            "FROM Messages m JOIN Conversations c ON (c.id = m.convo_id) "
            "WHERE 1 != 1 {}"
            "ORDER BY m.id ASC "
            "LIMIT {}".format(channels_query, messages_limit)
        )

        for row in result.fetchall():
            messages.append({
                "id": row[0],
                "author": row[1],
                "date": row[2],
                "content": row[3] if row[3] else "",
                "conversation": {"id": row[4], "name": row[5]}
            })

        connection.close()

        return messages
        
    def get_channels(self, limit):
        channels = []
 
        connection = sqlite3.connect(self.skype_database_temp)
        result = connection.cursor()
        result.execute(
            "SELECT id, displayname, last_activity_timestamp "
            "FROM Conversations "
            "WHERE last_activity_timestamp IS NOT NULL "
            "ORDER BY last_activity_timestamp DESC "
            "LIMIT {}".format(limit)
        )

        for row in result.fetchall():
            channels.append({
                "id": row[0],
                "display_name": format(row[1]),
                "last_activity": row[2]
            })

        connection.close()

        return channels